src/parameters.jl

```diff
@@ -4,10 +4,13 @@ Base.@kwdef mutable struct Parameters
     algorithm::String = "Line Search" # Defines the algorithm -- either Line Search or Trust Region Method (Trust Region algorithm hasn't been implemented yet)
 
     # Defines the external solver for suproblems
-    external_optimizer::Union{Nothing,DataType} = nothing
+    external_optimizer::Union{Nothing,DataType,MOI.OptimizerWithAttributes} = nothing
```

Comment:
Do we allow `DataType` type too? What if `external_optimizer` is given without attribute?

Reply:
That's why we should allow `DataType`. If we remove `DataType`, `external_optimizer` without `attributes` will give error.